Examples/FilterProgressReporting/Documentation.rst

```diff
@@ -5,6 +5,96 @@ Filter Progress Reporting
 Overview
 --------
 
+SimpleITK has the ability to add commands or callbacks as observers of
+events that may occur during data processing. This feature can be used to add
+progress reporting to a console, to monitor the process of optimization,
+to abort a process, or to improve the integration of
+SimpleITK into Graphical User Interface event queues.
+
+
+Events
+------
+
+Events are a simple enumerated type in SimpleITK, represented by
+the `EventEnum type <https://itk.org/SimpleITKDoxygen/html/namespaceitk_1_1simple.html#aa7399868984d99493c5a307cce373ace>`_.
+More information about each event type can be found in the documentation
+for the enum. All SimpleITK filters, including the reading
+and writing ones, are derived from the ProcessObject class which has
+support for events. SimpleITK utilizes the native ITK event system
+but has simpler events and methods to add an observer or commands. The goal
+is to provide a simpler interface more suitable for scripting
+languages.
+
+Commands
+--------
+
+The command design pattern is used to allow user code to be executed
+when an event occurs. It is encapsulated in the Command class. The
+Command class provides a virtual Execute method to be overridden in
+derived classes. Additionally, SimpleITK provides internal reference
+tracking between the ProcessObject and the Command. This reference tracking
+allows an object to be created on the stack or dynamically allocated, without
+additional burden.
+
+Command Directors for Wrapped Languages
+---------------------------------------
+SimpleITK uses SWIG's director feature to enable wrapped languages to
+derive classes from the Command class.  Thus a user may  override the
+Command class's Execute method for custom call-backs. The following
+languages support deriving classes from the Command class:
+
+Python
+......
+.. literalinclude:: FilterProgressReporting.py
+   :language: python
+   :start-after: [python director command]
+   :end-before: [python director command]
+
+Java
+....
+.. literalinclude:: FilterProgressReporting.java
+   :language: java
+   :start-after: sphinx_start_command_inheritance
```

Comment:
Please change this string to match Doxygen, by changing it to "java director command"